python/cudf_polars/cudf_polars/containers/column.py

```diff
@@ -439,28 +461,31 @@ def copy(self) -> Self:
             dtype=self.dtype,
         )
 
-    def mask_nans(self) -> Self:
+    def mask_nans(self, stream: Stream) -> Self:
         """Return a shallow copy of self with nans masked out."""
         if plc.traits.is_floating_point(self.obj.type()):
             old_count = self.null_count
-            mask, new_count = plc.transform.nans_to_nulls(self.obj)
+            mask, new_count = plc.transform.nans_to_nulls(self.obj, stream=stream)
             result = type(self)(self.obj.with_mask(mask, new_count), self.dtype)
             if old_count == new_count:
                 return result.sorted_like(self)
             return result
         return self.copy()
 
-    @functools.cached_property
-    def nan_count(self) -> int:
+    def nan_count(self, stream: Stream) -> int:
         """Return the number of NaN values in the column."""
-        if self.size > 0 and plc.traits.is_floating_point(self.obj.type()):
-            # See https://github.com/rapidsai/cudf/issues/20202 for we type ignore
-            return plc.reduce.reduce(
-                plc.unary.is_nan(self.obj),
-                plc.aggregation.sum(),
-                plc.types.SIZE_TYPE,
-            ).to_py()  # type: ignore[return-value]
-        return 0
+        if self._nan_count is None:
+            if self.size > 0 and plc.traits.is_floating_point(self.obj.type()):
+                # See https://github.com/rapidsai/cudf/issues/20202 for we type ignore
+                self._nan_count = plc.reduce.reduce(  # type: ignore[assignment]
+                    plc.unary.is_nan(self.obj, stream),
```

Comment:
nitpick: consistency
```suggestion
                    plc.unary.is_nan(self.obj, stream=stream),
```